cmd/kube-apiserver/app/server.go

```diff
@@ -574,7 +574,7 @@ func BuildStorageFactory(s *options.ServerRunOptions) (*serverstorage.DefaultSto
 		// FIXME (soltysh): this GroupVersionResource override should be configurable
 		[]schema.GroupVersionResource{
 			batch.Resource("cronjobs").WithVersion("v1beta1"),
-			storage.Resource("volumeattachments").WithVersion("v1alpha1"),
+			storage.Resource("volumeattachments").WithVersion("v1beta1"),
```

Comment:
What happens if we leave this `v1alpha1`? If we add new things to `v1alpha1` but not to the `v1beta1` will this handle it correctly?

Reply:
There could only be one storage version. You have to keep the v1alpha1 and v1beta1 roundtrippable.

If you want to test new things in alpha, you can create a new struct and specify here to store it as v1alpha1.

Reply:
If you store in v1alpha1, you have the option to downgrade and have VolumeAttachments keep working in alpha form, but you have alpha data in etcd, which means you cannot drop the alpha types until storage migration is solved. 

If you store in beta, you are better positioned to drop alpha types sooner (since the only people who would have alpha data were people who upgraded from 1.9 with the alpha flag enabled)

Reply:
I'd probably expect to store in v1beta1

Reply:
Ok. I'm slightly concerned that doing so will require us to add new features to both v1beta1 but v1alpha1, but, we'll leave storage as v1beta1.